src/ExadelMentorship.BusinessLogic/Services/MBus/ConnectionProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿using ExadelMentorship.BusinessLogic.Interfaces.MessageBus;
+using ExadelMentorship.BusinessLogic.Models;
+using Microsoft.Extensions.Options;
+using RabbitMQ.Client;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ExadelMentorship.BusinessLogic.Services.MBus
+{
+    public class ConnectionProvider : IConnectionProvider
+    {
+        private readonly ConnectionFactory _factory;
+
+        private readonly IConnection _connection;
+
+        public ConnectionProvider(IOptions<RabbitMQSettings> rabbitMQSettings)
+        {
+            _factory = new ConnectionFactory
+            {
+                Uri = new Uri(rabbitMQSettings.Value.Uri)
+            };
+            _connection = _factory.CreateConnection();
+        }
```

Comment:
You are not moving disposable (the connection) to DI. Read [Disposal of services](https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines)

[Injection Constructors should be simple](https://blog.ploeh.dk/2011/03/03/InjectionConstructorsshouldbesimple/). Move this out of constructor.

Becasue, [a class containing disposable field, must also implement IDisposable](https://www.google.com/search?client=firefox-b-d&q=class+with+disposable+field+must+be+also+disposable). Otherwise you are not a) disposing it and b) not giving anyone a way dispose it.

Reply:
simplified constructor, implemented IDisposable

Reply:
You made it even worse. The IDisposable part. It is bad in so many ways, that it is harder to explain.

Reply:
implemented IDisposable in correct way